src/test/unit_tests/jest_tests/test_nc_nsfs_connection_cli.test.js

```diff
@@ -0,0 +1,187 @@
+const { TMP_PATH, set_nc_config_dir_in_config } = require('../../system_tests/test_utils');
+const { TYPES, ACTIONS } = require('../../../manage_nsfs/manage_nsfs_constants');
+
+const tmp_fs_path = path.join(TMP_PATH, 'test_nc_nsfs_account_cli');
+const timeout = 5000;
+
+// eslint-disable-next-line max-lines-per-function
+describe('manage nsfs cli connection flow', () => {
+    describe('cli create connection', () => {
+        const config_root = path.join(tmp_fs_path, 'config_root_manage_nsfs');
+        const config_fs = new ConfigFS(config_root);
+        const root_path = path.join(tmp_fs_path, 'root_path_manage_nsfs/');
+        const defaults = {
+            name: 'conn1',
+            agent_request_object: {
+                host: "localhost",
+                "port": 9999,
+                "timeout": 100
+            },
+            request_options_object: {auth: "user:passw"},
+            notification_protocol: "http",
+
+        };
+        beforeEach(async () => {
+            await fs_utils.create_fresh_path(root_path);
+            set_nc_config_dir_in_config(config_root);
+        });
+
+        afterEach(async () => {
+            await fs_utils.folder_delete(`${config_root}`);
+            await fs_utils.folder_delete(`${root_path}`);
+        });
+
+        it('cli create connection ', async () => {
+
+            const options_file = path.join(TMP_PATH, "conn1.json");
+            fs.writeFileSync(options_file, JSON.stringify(defaults));
+
+            const action = ACTIONS.ADD;
+            const conn_options = { config_root, from_file: options_file };
+            await exec_manage_cli(TYPES.CONNECTION, action, conn_options);
+            const connection = await config_fs.get_connection_by_name(defaults.name);
+            assert_connection(connection, defaults, true);
+        }, timeout);
+
+
+        it('cli delete connection ', async () => {
+
+            const options_file = path.join(TMP_PATH, "conn1.json");
+            fs.writeFileSync(options_file, JSON.stringify(defaults));
+
+            const conn_options = { config_root, from_file: options_file };
+            await exec_manage_cli(TYPES.CONNECTION, ACTIONS.ADD, conn_options);
+            const connection = await config_fs.get_connection_by_name(defaults.name);
+            assert_connection(connection, defaults, true);
+
+            await exec_manage_cli(TYPES.CONNECTION, ACTIONS.DELETE, {config_root, name: defaults.name});
+            expect(fs.readdirSync(config_fs.connections_dir_path).filter(file => file.endsWith(".json")).length).toEqual(0);
+        }, timeout);
+
+        it('cli update connection ', async () => {
+
+            const options_file = path.join(TMP_PATH, "conn1.json");
+            fs.writeFileSync(options_file, JSON.stringify(defaults));
+
+            const conn_options = { config_root, from_file: options_file };
+            await exec_manage_cli(TYPES.CONNECTION, ACTIONS.ADD, conn_options);
```

Comment:
You can move these lines as a helper function for creating the connection (you need it in update, list, etc.).